extension/commands/testSmartContractCommand.ts

```diff
@@ -417,3 +479,174 @@ async function installNodeModules(dir: string, language: string): Promise<void>
+            if (content.indexOf(searchItem) > -1) {
+                requirements[obj][req].present = true;
+            } else {
+                needUpdate = true;
+            }
+        }
+    }
+    return needUpdate;
+}
+
+function getProperty(data: string, property: string, isGradle: boolean): string {
+    let start: number = -1 ;
+    let end: number = -1;
+    if (isGradle) {
+        // Find the requested property with it's children, ie, property { ... }.
+        // We therefore need to find the first curly bracket after the property,
+        // and the correspoding closing bracket. Then return that substring.
+        start = data.indexOf(property);
+        if (start !== -1) {
+            let count: number = 0;
+            for (let i: number = start + data.slice(start).indexOf('{'); i < data.length; i++) {
+                count += data[i] === '{' ? 1 : data[i] === '}' ? -1 : 0;
+                if (count === 0 && i > start) {
+                    end = i + 1;
+                    break;
+                }
+            }
+        }
+    } else {
+        start = data.indexOf(`<${property}>`);
+        end = data.indexOf(`</${property}>`) + `</${property}>`.length;
+    }
+
+    if (start === -1 || end === -1) {
+        throw new Error(`Could not find property '${property}'.`);
+    }
+    return data.slice(start, end);
+}
+
+function getReplacement(data: string, requirements: any, type: string, isGradle: boolean): string {
+    for (const obj in requirements[type]) {
+        if (!requirements[type][obj].present) {
+            if (requirements[type][obj].url) {
+                if (isGradle) {
+                    data = data.replace(/\n*}$/, '\n}').replace(/}$/, `\tmaven {\n\t\turl "${requirements[type][obj].url}"\n\t}\n}`);
+                } else {
+                    data = data.replace(/\n*\s*<\/repositories>/, `\n\t\t<repository>\n\t\t\t<id>${requirements[type][obj].id}</id>\n\t\t\t<url>${requirements[type][obj].url}</url>\n\t\t</repository>\n\t</repositories>`);
+                }
+            } else {
+                if (isGradle) {
+                    data = data.replace(/\n*}$/, '\n}').replace(/}$/, `\ttestImplementation '${requirements[type][obj].groupId}:${requirements[type][obj].artifactId}:${requirements[type][obj].version}'\n}`);
+                } else {
+                    data = data.replace(/\n*\s*<\/dependencies>/, `\n\t\t<dependency>\n\t\t\t<groupId>${requirements[type][obj].groupId}</groupId>\n\t\t\t<artifactId>${requirements[type][obj].artifactId}</artifactId>\n\t\t\t<version>${requirements[type][obj].version}</version>\n\t\t\t<scope>test</scope>\n\t\t</dependency>\n\t</dependencies>`);
+                }
+
+            }
+        }
+    }
+    return data;
+}
+
+async function mvOrRmBuildFileCopy(packageFolder: vscode.WorkspaceFolder, remove: boolean): Promise<void> {
+    const gradleFileTmp: string = `${packageFolder.uri.fsPath}/build.gradle.tmp`;
```

Comment:
I think we should probably do `path.join(packageFolder.uri.fsPath, 'build.gradle.tmp')` to prevent any OS file-system errors.

(Same for line 638 - mavenFileTmp)